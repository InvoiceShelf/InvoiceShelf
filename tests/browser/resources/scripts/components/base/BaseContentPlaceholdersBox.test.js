import { h } from 'vue'
import { expect, test } from 'vitest'
import { render } from 'vitest-browser-vue'
import BaseContentPlaceholders from '@/scripts/components/base/BaseContentPlaceholders.vue'
import BaseContentPlaceholdersBox from '@/scripts/components/base/BaseContentPlaceholdersBox.vue'

test('BaseContentPlaceholdersBox animated, rounded, circle', async () => {
  const { getByTestId } = render(BaseContentPlaceholders, {
    props: {
      animated: false,
      'data-testid': 'box-placeholder',
      'class': 'flex flex-col gap-y-4'
    },
    slots: {
      default: () => [
        h(
          BaseContentPlaceholdersBox,
          {
            rounded: true,
            animated: true,
            'class': 'w-full h-24'
          }
        ),
        h(
          BaseContentPlaceholdersBox,
          {
            circle: true,
            'class': 'w-5 h-5'
          }
        ),
        h(
          BaseContentPlaceholdersBox,
          {
            rounded: true,
            'class': 'w-full h-24'
          }
        ),

      ]
    }
  })
  const boxPlaceholder = getByTestId('box-placeholder')
  await expect.element(boxPlaceholder).toBeInTheDocument()
  expect(boxPlaceholder).toMatchSnapshot()
})
