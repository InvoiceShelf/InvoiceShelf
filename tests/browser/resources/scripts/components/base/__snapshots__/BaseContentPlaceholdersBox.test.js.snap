// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseContentPlaceholdersBox animated, rounded, circle 1`] = `
PlaywrightLocator {
  "_container": <body>
    
    
    
  


    <div>
      <div
        class="base-content-placeholders flex flex-col gap-y-4"
        data-testid="box-placeholder"
      >
        
        <div
          class="base-content-placeholders-box base-content-placeholders-is-rounded base-content-placeholders-is-animated w-full h-24"
        />
        <div
          class="base-content-placeholders-box base-content-circle w-5 h-5"
        />
        <div
          class="base-content-placeholders-box base-content-placeholders-is-rounded w-full h-24"
        />
        
      </div>
    </div>
  </body>,
  "_parsedSelector": {
    "capture": undefined,
    "parts": [
      {
        "body": [
          {
            "simples": [
              {
                "combinator": "",
                "selector": {
                  "css": "body",
                  "functions": [],
                },
              },
            ],
          },
        ],
        "name": "css",
        "source": "body",
      },
      {
        "body": "[data-testid="box-placeholder"s]",
        "name": "internal:testid",
        "source": "[data-testid="box-placeholder"s]",
      },
    ],
  },
  "_pwSelector": undefined,
  "selector": "body >> internal:testid=[data-testid="box-placeholder"s]",
}
`;
